src/services/codefixes/inferFromUsage.ts

```diff
@@ -148,6 +152,11 @@ namespace ts.codefix {
             containingFunction.parameters.map(p => isIdentifier(p.name) ? inferTypeForVariableFromUsage(p.name, sourceFile, program, cancellationToken) : undefined);
         if (!types) return undefined;
 
+        // We didn't actually find a set of type inference positions matching each parameter position
+        if (containingFunction.parameters.length !== types.length) {
+            return undefined;
+        }
+        
         const textChanges = arrayFrom(mapDefinedIterator(zipToIterator(containingFunction.parameters, types), ([parameter, type]) =>

```

Comment:
Now that we are doing the `length` check above, can we unroll this to an index-based for-loop? the intermediate iterator and maps are unnecessary